confd/pkg/backends/calico/routes.go

```diff
@@ -166,43 +177,47 @@ func (rg *routeGenerator) getServiceForEndpoints(ep *v1.Endpoints) (*v1.Service,
 }
 
 // getEndpointsForService retrieves the corresponding ep for the given svc
-func (rg *routeGenerator) getEndpointsForService(svc *v1.Service) (*v1.Endpoints, string) {
-	// get key
+func (rg *routeGenerator) getEndpointsForService(svc *v1.Service) ([]*discoveryv1.EndpointSlice, string) {
+	var eps []*discoveryv1.EndpointSlice
+	for _, obj := range rg.epIndexer.List() {
```

Comment:
This is a complexity upgrade - going from O(1) to O(#eps) is probably OK, but the performance of this code now depends on roughly the number of services in the cluster.

We could keep a mapping of service key -> endpointslice keys (and vice-versa) if we wanted to address this. Probably not needed for this PR but might be worth looking into as a follow-up, if it's not too complex.